src/test/java/TestGUIMainFrame.java

```diff
@@ -35,6 +40,51 @@ void testJTableCreation() {
 		String[] header = {"header1", "header2"};
 		String[][] content = {{"cell0",  "cell1"},{"cell2", "cell3"}};
 		guiWithTable = new MainFrame(new JTable(content, header), MainController.getMainControllerInstance().getRulesList());
+		assertEquals(2, guiWithTable.getExcelTable().getColumnCount());
+	}
+	
+	/**
+	 * Tests if after MainFrame construction the buttons were created and are not null
+	 */
+	@Test
+	final void testGetCheckQualityButton() {
+		assertNotNull(frame.getCheckQualityButton());
+		assertNotNull(frame.getEditButton());
+		assertNotNull(frame.getAddButton());
+	}
+	
+	/**
+	 * Tests getFrameWidth method
+	 */
+	@Test
+	final void testGetFrameWidth() {
+		assertEquals(700, frame.getFrameWidth());
 	}
 
+	/**
+	 * Test getRulesComboBox method by checking if its created
+	 */
+	@Test
+	final void testGetRulesComboBox() {
+		assertNotNull(frame.getRulesComboBox());

```

Comment:
does this not fail?

Reply:
Nop, @BeforeEach creates the frame that creates a new RulesComboBox